src/Features/Tas/TasTools.cpp

```diff
@@ -0,0 +1,148 @@
+#include "TasTools.hpp"
+
+#include <cmath>
+
+#include "Features/Session.hpp"
+#include "Modules/Client.hpp"
+#include "Modules/Engine.hpp"
+#include "Modules/Server.hpp"
+
+#include "Utils/Math.hpp"
+#include "Utils/SDK.hpp"
+#include <Offsets.hpp>
+
+TasTools* tasTools;
+
+TasTools::TasTools()
+{
+    this->hasLoaded = true;
+    this->m_previous_speed = Vector{ 0, 0, 0 };
+    this->m_last_tick = (session->isInSession) ? engine->GetSessionTick() : 0;
```

Comment:
This will always be initialized to 0 since session gets initialized at the same time as this